<?xml version="1.0"?>
<launch>

	<!-- topic name -->
	<arg name="ndt_pose_topic_name" default="/ndt_pose" />
	<arg name="imu_topic_name" default="/sensing/imu/gx5/imu/data" />
	<arg name="ekf_pose_topic_name" default="/ekf_pose" />
	<arg name="odom_topic_name" default="/atom/odometry" />
	<arg name="velodyne_topic_name" default="/hesai/pandar" />
	<arg name="map_topic_name" default="/global_map" />
	<arg name="ndt_pc_topic_name" default="/ndt_pc" />

	<!-- <arg name="pcd_path" default="/home/ubuntu/Downloads/0.15_map_all.pcd" /> -->

	<arg name="respawn_pose_topic_name" default="/position/respawn" />

	<!-- frame_id -->
	<arg name="map_frame_id" default="map" />
	<arg name="odom_frame_id" default="odometry" />
	<arg name="base_link_frame_id" default="base_link" />
	<arg name="camera_frame_id" default="camera_link" />
	<arg name="velodyne_frame_id" default="Pandar" />

	<!-- ekf -->
	<node pkg="ekf_localizer" type="ekf_node" name="ekf_node" >
		<param name="ndt_pose_topic_name" value="$(arg ndt_pose_topic_name)" />
		<param name="imu_topic_name" value="$(arg imu_topic_name)" />
		<param name="ekf_pose_topic_name" value="$(arg ekf_pose_topic_name)" />
		<param name="odom_topic_name" value="$(arg odom_topic_name)" />
		<param name="map_frame_id" value="$(arg map_frame_id)" />
		<param name="odom_frame_id" value="$(arg odom_frame_id)" />
		<param name="base_link_frame_id" value="$(arg base_link_frame_id)" />
		<param name="is_3DoF" value="true" />
		<param name="is_odom_tf" value="false" />

		<param name="respawn_pose_topic_name" value="$(arg respawn_pose_topic_name)" />

		<rosparam command="load" file="$(find ekf_localizer)/config/ekf/ekf.yaml" />
	</node>


	<!-- map_matcher -->
	<node pkg="ekf_localizer" type="map_matcher_node" name="map_matcher_node" >
		<!-- <param name="pcd_file_path" value="$(arg pcd_path)" /> -->
		<param name="pcd_file_path" value="/home/ubuntu/Downloads/0.15_map_all.pcd" />
		<param name="pc_topic_name" value="$(arg velodyne_topic_name)" />
		<param name="orb_pc_topic_name" value="" />
		<param name="ekf_pose_topic_name" value="$(arg ekf_pose_topic_name)" />
		<param name="ndt_pose_topic_name" value="$(arg ndt_pose_topic_name)" />
		<param name="map_topic_name" value="$(arg map_topic_name)" />
		<param name="ndt_pc_topic_name" value="$(arg ndt_pc_topic_name)" />
		<param name="map_frame_id" value="$(arg map_frame_id)" />
		<param name="is_publish_map" value="true" />
		<param name="is_pcl_offset" value="false" />
		<rosparam command="load" file="$(find ekf_localizer)/config/map_matcher/map_matcher.yaml" />
	</node>

	<!-- tf -->
	<!-- <node pkg="ekf_localizer" type="tf_broadcaster_node" name="tf_broadcaster_node" >
		<param name="is_odom_tf" value="true" />
		<param name="odom_topic_name" value="$(arg odom_topic_name)" />
		<param name="odom_frame_id" value="$(arg odom_frame_id)" />
		<param name="base_link_frame_id" value="$(arg base_link_frame_id)" />
		<param name="camera_frame_id" value="$(arg camera_frame_id)" />
		<param name="velodyne_frame_id" value="$(arg velodyne_frame_id)" />
		<rosparam command="load" file="$(find ekf_localizer)/config/tf/ikuta.yaml" />
	</node> -->


	<!-- <node pkg="tf" type="static_transform_publisher" name="kinect_base_link5" args="0 0 0 0 0 0 /map /velodyne_lower 100"/> -->
	 <!-- <node name="map_to_lidar" pkg="tf2_ros" type="static_transform_publisher" args="0 0 0 0 0 0 0 map velodyne_lower"/> -->
	<!-- rviz -->
	<node pkg="rviz" type="rviz" name="rviz" args="-d $(find ekf_localizer)/config/ekf_localizer.rviz" />	
</launch>
